example/dice/main.go

```diff
@@ -0,0 +1,93 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func run() (err error) {
+	// Handle CTRL+C gracefully.
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	// Setup OpenTelemetry.
+	serviceName := "dice"
+	serviceVersion := "0.1.0"
+	otelShutdown, err := setupOTelSDK(ctx, serviceName, serviceVersion)
+	if err != nil {
+		return
+	}
+	// Handle shutdown properly so nothing leaks.
+	defer func() {
+		err = errors.Join(err, otelShutdown(context.Background()))
```

Comment:
This error seems to never be handled?

Reply:
Named return 😉

Reply:
Yes, but the `return` (on line 84) specifies its own error, it doesn't use the named variable.

Reply:
It will work. See:  https://go.dev/play/p/QM3oLhEKkTl

But maybe it will be better to refactor line 84 to:

```go
	err = srv.Shutdown(context.Background())
```

WDYT?

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/4539/commits/d87f3431b52f5f404e2318c89e25dbe46736d227

Reply:
Your latest commit makes it easier indeed, though I think it's a bit too magical (not to have things happening in-line).

Reply:
What do you propose instead? I do not think there is any other option than using "named returns" to alter the error value returned in defer statement.

Reply:
> And the return would be errors.Join(errs...)?

The returned `errors.Join(errs...)` would still miss the error returned by deferred shutdown.

Reply:
For reference: https://wstrm.dev/posts/errors-join-heart-defer/